### PR TITLE
feat: libevse-security will try to fix malformed PEM strings

### PR DESCRIPTION
**Pull request by valentin-dimov**

### CODE REVIEW

File: include/evse_security/utils/fix_pem_string.hpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+#include <boost/algorithm/string.hpp>
+#include <boost/algorithm/string/regex.hpp>
+
+class MalformedPEMException: public std::runtime_error {
+    using std::runtime_error::runtime_error;
+};
+
+static std::string fix_pem_payload(const std::string& input_pem_payload) {
+    // Remove all newlines and whitespace from the rest
+    std::string cleared_string = boost::erase_all_regex_copy(input_pem_payload, boost::regex("[^a-zA-Z0-9/\\+=]"));
```
Comment (hikinggrass) [inline include/evse_security/utils/fix_pem_string.hpp:16]: Do we need boost(algorithm/regex) here?

Comment (valentin-dimov) [reply include/evse_security/utils/fix_pem_string.hpp]: We don't, removed it.

File: lib/evse_security/evse_security.cpp [hunk 2]
```diff
@@ -152,9 +154,13 @@ EvseSecurity::~EvseSecurity() {
 InstallCertificateResult EvseSecurity::install_ca_certificate(const std::string& certificate,
                                                               CaCertificateType certificate_type) {
     EVLOG_debug << "Installing ca certificate: " << conversions::ca_certificate_type_to_string(certificate_type);
+
+    // Some CSMS may send malformed PEM strings. Canonicalize them here.
```
Comment (Pietfried) [inline lib/evse_security/evse_security.cpp:158]: nit: I would not refer `CSMS` here; could be used by any consumer of the lib

```suggestion
    // Function may be called with malformed PEM strings. Canonicalize them here.
```

File: lib/evse_security/evse_security.cpp [hunk 3]
```diff
@@ -699,6 +705,9 @@ bool EvseSecurity::verify_file_signature(const fs::path& path, const std::string
                                          const std::string signature) {
     EVLOG_debug << "Verifying file signature for " << path.string();
 
+    // Make sure the signing certificate is proper PEM
+    std::string fixed_signing_certificate = fix_pem_string(signing_certificate);
```
Comment (Pietfried) [inline lib/evse_security/evse_security.cpp:709]: nit:

```suggestion
    const auto fixed_signing_certificate = fix_pem_string(signing_certificate);
```

File: lib/evse_security/evse_security.cpp [hunk 4]
```diff
@@ -152,9 +154,13 @@ EvseSecurity::~EvseSecurity() {
 InstallCertificateResult EvseSecurity::install_ca_certificate(const std::string& certificate,
                                                               CaCertificateType certificate_type) {
     EVLOG_debug << "Installing ca certificate: " << conversions::ca_certificate_type_to_string(certificate_type);
+
+    // Some CSMS may send malformed PEM strings. Canonicalize them here.
+    std::string fixed_certificate = fix_pem_string(certificate);
```
Comment (Pietfried) [inline lib/evse_security/evse_security.cpp:159]: nit:
```suggestion
    const auto fixed_certificate = fix_pem_string(certificate);
```

**Review by Pietfried — APPROVED** [review]

**Pull request merged**